### PR TITLE
clpm: remove at 0.4.1

### PR DESCRIPTION
**Pull request by philiptaron**
## Description of changes

[It doesn't build](https://hydra.nixos.org/job/nixos/trunk-combined/nixpkgs.clpm.x86_64-linux), there hasn't been a release in two years, and attempting to upgrade to the most recent tag (`v0.4.2-rc.2`) reveals that it immediately returns exit code 1 without output.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by storvik — APPROVED** [review]
I agree, we should just remove this.

File: pkgs/development/compilers/sbcl/default.nix [hunk 1]
```diff
@@ -19,11 +19,6 @@
 
 let
   versionMap = {
-    # Only kept around for BCLM. Remove once unneeded there.
-    "2.1.9" = {
-      sha256 = "189gjqzdz10xh3ybiy4ch1r98bsmkcb4hpnrmggd4y2g5kqnyx4y";
-    };
-
     "2.4.2" = {
```
Comment (hraban) [inline pkgs/development/compilers/sbcl/default.nix:22]: love it. Can also set coreCompression to `true` and markRegionGC to `true` in default args.

Comment (philiptaron) [reply pkgs/development/compilers/sbcl/default.nix:22]: Wonderful news. I'll leave that to a future PR from someone that understands SBCL a little more. 🧷 

I'm just a curious lad looking for broken or odd packages.

Comment (hraban) [reply pkgs/development/compilers/sbcl/default.nix:22]: ... I meant at the top of this file 😂

**Pull request merged**